streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -83,7 +84,12 @@ import org.apache.spark.storage.StorageLevel
  * }}}
  */
 @DeveloperApi
-abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable {
+abstract class Receiver[T](val storageLevel: StorageLevel)(implicit t: ClassTag[T])
```

Comment:
I see, so in that case we should carefully think about this change, at least keep compatible with existing code.